Integrations/integration-FreshDesk.yml

```diff
@@ -0,0 +1,2574 @@
+        """
+        Determine if the value points to a group by checking against all group names and IDs, and adjust
+        'args' accordingly.
+        """
+        assigned_group = args.get(key_name, None)
+        if assigned_group:
+            group_names, group_ids = [], []
+            # Get names and ids of groups
+            groups = list_groups()
+            for group in groups:
+                group_ids.append(group.get('id'))
+                group_names.append(group.get('name').lower())
+            # Check if responder value is a group ID
+            if assigned_group in group_ids:
+                args['group_id'] = assigned_group
+                del args[key_name]
+            # Or the name of a group
+            elif assigned_group.lower() in group_names:
+                args['group_id'] = group_ids[group_names.index(assigned_group.lower())]
+                del args[key_name]
+        return args
+
+
+    def http_request(method, url_suffix, params=None, data=None, files=None,):
+        """
+        A wrapper for requests lib to send our requests and handle requests and responses better.
+        """
+        res = requests.request(
+            method,
+            BASE_URL + url_suffix,
+            verify=USE_SSL,
+            params=params,
+            data=data,
+            headers=HEADERS,
+            auth=AUTH,
+            files=files,
+        )
+        # Handle error responses gracefully
+        try:
+            if res.status_code not in [200, 201, 202, 204]:
+                LOG(res.json())
+                LOG(res.text)
+                LOG.print_log()
+                err_msg = 'Error in API call to Freshdesk Integration [{}] - {}'.format(res.status_code, res.reason)
+                err = json.loads(res.content)
+                if err.get('errors'):
+                    for error in err.get('errors'):
+                        err_msg += '\n' + json.dumps(error, indent=2)
+                else:
+                    for key, value in res.json().iteritems():
+                        err_msg += '\n{}: {}'.format(key, value)
+                return_error(err_msg)
+            # Handle response with no content
+            elif res.status_code == 204:
+                return res
+
+            return res.json()
+        except Exception as e:
+            LOG(e)
+            ex_type, ex, tb = sys.exc_info()
+            traceback.print_tb(tb)
+            return_error('error occurred in the http_request method')
+
+
+    ''' COMMANDS + REQUESTS FUNCTIONS '''
+
+
+    def test_module():
+        """
+        Will try to make a request to the API endpoint for listing all tickets.
+        """
+        response = http_request('GET', 'tickets')
+
+
+    def fetch_incidents():
+        # demisto.getLastRun() will returns an obj with the previous run in it.
+        last_run = demisto.getLastRun()
+        # Get the last fetch time, if exists
+        last_fetch = last_run.get('time')
```

Comment:
is this the best way to handle this? can't we check if the response empty in another way ?